File: src/main/java/com/leijendary/spring/boot/template/model/UUIDModel.java
package com.leijendary.spring.boot.template.model;

import java.util.UUID;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@MappedSuperclass
public abstract class UUIDModel extends AppModel {

    @Id
    @GeneratedValue
    private UUID id;
}

File: src/main/kotlin/com/leijendary/spring/template/core/model/UUIDModel.kt
package com.leijendary.spring.template.core.model

import com.leijendary.spring.template.core.projection.UUIDProjection
import jakarta.persistence.Column
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.util.*
import java.util.UUID.randomUUID

@MappedSuperclass
abstract class UUIDModel : UUIDProjection {
    @Id
    @GeneratedValue
    @Column(updatable = false)
    override var id: UUID = randomUUID()

    @Column(insertable = false, updatable = false)
    var rowId: Long? = null
}
